src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -379,20 +381,22 @@ class SamRecordClipper(val mode: ClippingMode, val autoClipAttributes: Boolean)
     * @return the additional number of bases clipped (3' end in sequencing order)
     */
   def clipExtendingPastMateEnd(rec: SamRecord): Int = {
-    val mateEnd = rec.mateEnd.getOrElse(rec.start + abs(rec.insertSize) - 1)
-    clipExtendingPastMateEnd(rec=rec, mateEnd=mateEnd)
+    val mateUnclippedStart = rec.mateUnclippedStart.getOrElse(rec.mateStart)
+    val mateUnclippedEnd   = rec.mateUnclippedEnd.getOrElse(rec.start + abs(rec.insertSize) - 1)
```

Comment:
This function's documentation describes how the mate end is calculated but needs to also have a similar description for how we are calculating the mate start. 

Also, given how important the `MC` tag is to this subtle behavior change, I'm thinking we should fixup the `ClipBam` tool help message to say that the `MC` tag really should be present (and how to add it if it does not exist!).

Also, there are no unit tests covering this functionality according to codecov. Could you add some?

Reply:
EDIT: on second thought, since there are no unit tests covering this function I suspect this function is not really used. Can we delete it? I am worrisome it will be used accidentally without the `MC` tag present and the data will not be optimal. I have a similar concern about the `numBasesExtendingPastMate` that infers the mate's start/end (biasing towards the unclipped positions).